fineract-provider/src/main/java/org/apache/fineract/portfolio/note/domain/Note.java

```diff
@@ -37,23 +42,27 @@
 import org.apache.fineract.portfolio.shareaccounts.domain.ShareAccount;
 
 @Entity
+@Getter
+@NoArgsConstructor
+@AllArgsConstructor
```

Comment:
AllArgsConstructor does not make sense for this entity, because from the several entity-type references only one (or in some special cases maybe more but never all) can be set.
The Builder will generate a private AllArgsConstructor anyway. Could you please remove this annotation?

Reply:
Actually, the `@Builder` annotation requires a proper constructor for initializing the entity, so I've limited the access to private instead.

Reply:
If a class is annotated with @Builder, then a package-private constructor is generated with all fields as arguments (as if @AllArgsConstructor(access = AccessLevel.PACKAGE) is present on the class), and it is as if this constructor has been annotated with @Builder instead. Note that this constructor is only generated if you haven't written any constructors and also haven't added any explicit @XArgsConstructor annotations. In those cases, lombok will assume an all-args constructor is present and generate code that uses it; this means you'd get a compiler error if this constructor is not present.
So yes, you need to add the @AllArgsConstructor, just because @NoArgsConstructor is there. Private is fine.